pynamodb/models.py

```diff
@@ -1009,7 +1009,7 @@ def _get_connection(cls) -> TableConnection:
                     cls.__module__, cls.__name__,
                 ),
             )
-        if cls._connection is None:
+        if cls._connection is None or cls._connection.table_name != cls.Meta.table_name:
```

Comment:
Worth leaving a comment that the connection should be recreated (or updated) if any of these values change? It's also fine to do this in a future PR